File: src/api/git/lfs.ts
import { invoke } from '@tauri-apps/api/core';

export interface LfsLockInfo {
  id: string;
  path: string;
  owner: { name: string } | null;
  locked_at: string;
}

export async function trackLfsPatterns(patterns: string[]): Promise<void> {
  await invoke('lfs_track_patterns', { patterns });
}

export async function listTrackedLfsPatterns(): Promise<string[]> {
  return await invoke('lfs_list_tracked_patterns');
}

export async function listLfsLocks(): Promise<LfsLockInfo[]> {
  return await invoke('lfs_list_locks');
}

export async function pushLfs(remote: string, branch: string): Promise<string> {
  return await invoke('lfs_push', { remote, branch });
}

File: src-tauri/src/commands/lfs.rs
use crate::git;
use crate::state::AppState;
use tauri::State;

#[tauri::command]
pub fn lfs_track_patterns(
    patterns: Vec<String>,
    state: State<'_, AppState>,
) -> crate::error::Result<()> {
    let repo_path = state.repo_path.lock().unwrap();
    if let Some(p) = repo_path.as_ref() {
        git::lfs::track_patterns(p, patterns)
    } else {
        Err(crate::error::GitError::CommandFailed(
            "No repository is currently open.".to_string(),
        ))
    }
}

#[tauri::command]
pub fn lfs_list_tracked_patterns(state: State<'_, AppState>) -> crate::error::Result<Vec<String>> {
    let repo_path = state.repo_path.lock().unwrap();
    if let Some(p) = repo_path.as_ref() {
        git::lfs::list_tracked_patterns(p)
    } else {
        Err(crate::error::GitError::CommandFailed(
            "No repository is currently open.".to_string(),
        ))
    }
}

#[tauri::command]
pub fn lfs_list_locks(
    state: State<'_, AppState>,
) -> crate::error::Result<Vec<git::lfs::LfsLockInfo>> {
    let repo_path = state.repo_path.lock().unwrap();
    if let Some(p) = repo_path.as_ref() {
        git::lfs::list_locks(p)
    } else {
        Err(crate::error::GitError::CommandFailed(
            "No repository is currently open.".to_string(),
        ))
    }
}

#[tauri::command]
pub fn lfs_push(
    remote: String,
    branch: String,
    state: State<'_, AppState>,
) -> crate::error::Result<String> {
    let repo_path = state.repo_path.lock().unwrap();
    if let Some(p) = repo_path.as_ref() {
        git::lfs::push_lfs(p, &remote, &branch)
    } else {
        Err(crate::error::GitError::CommandFailed(
            "No repository is currently open.".to_string(),
        ))
    }
}
